kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaAuthnHandler.java

```diff
@@ -387,7 +388,10 @@ private void doTransition(Channel channel, State next) {
 
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
-        if (msg instanceof BareSaslRequest) {
+        if (isInitialHaProxyMessage(msg)) {
+            ctx.fireChannelRead(msg);
```

Comment:
I think this should be configurable. Ordinarily the upstream will be Apache Kafka which doesn't understand the HA Proxy protocol.   So this should be something the user has to be enable.